Sources/CacheStore/Stores/Store/Store.swift

```diff
@@ -113,8 +113,8 @@ open class Store<Key: Hashable, Action, Dependency>: ObservableObject, ActionHan
     /// Sends the action to be handled by the `Store`
     public func handle(action: Action) {
         guard Thread.isMainThread else {
-            DispatchQueue.main.async { [weak self] in
-                self?.handle(action: action)
+            DispatchQueue.main.async {
+                self.handle(action: action)
             }
```

Comment:
`await MainActor.run { self.handle(action: action)` 

Might be a good place to introduce this if possible?

Reply:
It would be if this function was async capable, but it would need to be wrapped in a `Task { await MainActor.run { self.handle(action: action) }` to work in the current context...

Reply:
Ah that's right! I didn't see that!